From 0a185632e91fd83d3a803dc91ef1209cf17cbff8 Mon Sep 17 00:00:00 2001
From: Wang Panzhenzhuan <randy.wang@rock-chips.com>
Date: Tue, 18 Sep 2018 08:59:20 +0800
Subject: [PATCH] arm64: dts: rockchip: move camera node config to dts for
 rk3326-863-lp3-v10

Move rkisp1 and gc0312/gc2145, cif and cif_sensor
node config to dts level not in dtsi.

For rkisp1 and gc0312/gc2145 don't need by rk3326-863-lp3-v10.dts,
otherwise it causes 8.0 863 machine camera register failed;
And cif with cif_sensor don't need by rk3326-863-lp3-v10-avb.dts,
otherwise it causes 9.0 863 machine camera register failed;

So need to modify rk3326-863-lp3-v10-avb.dts and
rk3326-863-lp3-v10.dts separately for camera node configs

Change-Id: Ieac54c39cec1f2eae70e0d41829cc0ea1b6a3eb2
Signed-off-by: Wang Panzhenzhuan <randy.wang@rock-chips.com>
---
 .../dts/rockchip/rk3326-863-lp3-v10-avb.dts   | 81 +++++++++++++++++++
 .../boot/dts/rockchip/rk3326-863-lp3-v10.dts  |  8 ++
 .../boot/dts/rockchip/rk3326-863-lp3-v10.dtsi | 72 -----------------
 3 files changed, 89 insertions(+), 72 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10-avb.dts b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10-avb.dts
index 2b69d08f63af..ce9dc36787dd 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10-avb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10-avb.dts
@@ -28,3 +28,84 @@
 		};
 	};
 };
+
+&i2c2 {
+	status = "okay";
+
+	gc0312@21 {
+		status = "okay";
+		compatible = "galaxycore,gc0312";
+		reg = <0x21>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&cif_clkout_m0>;
+
+		clocks = <&cru SCLK_CIF_OUT>;
+		clock-names = "xvclk";
+
+		avdd-supply = <&vcc2v8_dvp>;
+		dovdd-supply = <&vcc1v8_dvp>;
+		dvdd-supply = <&vcc1v8_dvp>;
+
+		pwdn-gpios = <&gpio2 14 GPIO_ACTIVE_HIGH>;
+
+		port {
+			gc0312_out: endpoint {
+				remote-endpoint = <&dvp_in_fcam>;
+			};
+		};
+	};
+
+	gc2145@3c {
+		status = "okay";
+		compatible = "galaxycore,gc2145";
+		reg = <0x3c>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&cif_clkout_m0>;
+
+		clocks = <&cru SCLK_CIF_OUT>;
+		clock-names = "xvclk";
+
+		avdd-supply = <&vcc2v8_dvp>;
+		dovdd-supply = <&vcc1v8_dvp>;
+		dvdd-supply = <&vcc1v8_dvp>;
+
+		pwdn-gpios = <&gpio2 13 GPIO_ACTIVE_HIGH>;
+
+		port {
+			gc2145_out: endpoint {
+				remote-endpoint = <&dvp_in_bcam>;
+			};
+		};
+	};
+};
+
+&isp_mmu {
+	status = "okay";
+};
+
+&rkisp1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&cif_clkout_m0 &dvp_d0d1_m0 &dvp_d2d9_m0 &dvp_d10d11_m0>;
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dvp_in_fcam: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&gc0312_out>;
+			};
+
+			dvp_in_bcam: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&gc2145_out>;
+			};
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
index 37ef6af0b5ac..7a399b3356ac 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dts
@@ -12,6 +12,14 @@
 	compatible = "rockchip,rk3326-863-lp3-v10", "rockchip,rk3326";
 };
 
+&cif {
+	status = "okay";
+};
+
+&cif_sensor {
+	status = "okay";
+};
+
 &firmware_android {
 	compatible = "android,firmware";
 	fstab {
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dtsi
index 270acbad5e73..0a2330907d8a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3326-863-lp3-v10.dtsi
@@ -156,78 +156,6 @@
 	};
 };
 
-&i2c2 {
-	status = "okay";
-
-	gc0312@21 {
-		status = "okay";
-		compatible = "galaxycore,gc0312";
-		reg = <0x21>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&cif_clkout_m0>;
-
-		clocks = <&cru SCLK_CIF_OUT>;
-		clock-names = "xvclk";
-
-		avdd-supply = <&vcc2v8_dvp>;
-		dovdd-supply = <&vcc1v8_dvp>;
-		dvdd-supply = <&vcc1v8_dvp>;
-
-		pwdn-gpios = <&gpio2 14 GPIO_ACTIVE_HIGH>;
-
-		port {
-			gc0312_out: endpoint {
-				remote-endpoint = <&dvp_in_fcam>;
-			};
-		};
-	};
-
-	gc2145@3c {
-		status = "okay";
-		compatible = "galaxycore,gc2145";
-		reg = <0x3c>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&cif_clkout_m0>;
-
-		clocks = <&cru SCLK_CIF_OUT>;
-		clock-names = "xvclk";
-
-		avdd-supply = <&vcc2v8_dvp>;
-		dovdd-supply = <&vcc1v8_dvp>;
-		dvdd-supply = <&vcc1v8_dvp>;
-
-		pwdn-gpios = <&gpio2 14 GPIO_ACTIVE_HIGH>;
-
-		port {
-			gc2145_out: endpoint {
-				remote-endpoint = <&dvp_in_bcam>;
-			};
-		};
-	};
-};
-
-&rkisp1 {
-	status = "okay";
-
-	pinctrl-names = "default";
-	pinctrl-0 = <&cif_clkout_m0 &dvp_d0d1_m0 &dvp_d2d9_m0 &dvp_d10d11_m0>;
-
-	ports {
-		port@0 {
-			dvp_in_fcam: endpoint@0 {
-				remote-endpoint = <&gc0312_out>;
-			};
-
-			dvp_in_bcam: endpoint@1 {
-				remote-endpoint = <&gc2145_out>;
-			};
-		};
-	};
-};
-
-&isp_mmu {
-	status = "okay";
-};
 
 &display_subsystem {
 	status = "okay";
-- 
2.35.3

